h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -233,6 +233,9 @@ private void validateGamParameters() {
         if (!dataset.vec(cname).isNumeric())
           error("gam_columns", "column " + cname + " is not numerical and cannot be used as a gam" +
                   " column.");
+        if (dataset.vec(cname).isInt() && ((dataset.vec(cname).max() - dataset.vec(cname).min() + 1) < _parms._num_knots[index]))
+          error("gam_columns", "column " + cname + " has cardinality lower than the number of knots and cannot be used as a gam" +
+                  " column.");
       }
```

Comment:
This is a good first order check.  However, even if this check passes, failures can still occur if say we have 100 rows of 1 and 100 rows of 100 and numknot = 1.  It is necessary but not sufficient.  Since you kept the old verification check, it should not be a problem.